src/Altinn.Platform/Altinn.Platform.Events/Tests/TestingServices/RegisterServiceTest.cs

```diff
@@ -0,0 +1,151 @@
+            _handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            _contextAccessor = new Mock<IHttpContextAccessor>();
+            _accessTokenGenerator = new Mock<IAccessTokenGenerator>();
+        }
+
+        [Fact]
+        public async Task PartyLookup_MatchFound_IdReturned()
+        {
+            // Arrange
+            Party party = new Party
+            {
+                PartyId = 500000,
+                OrgNumber = "897069650",
+            };
+            int expected = 500000;
+            HttpResponseMessage httpResponseMessage = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(JsonSerializer.Serialize(party), Encoding.UTF8, "application/json")
+            };
+
+            HttpRequestMessage actualRequest = null;
+            void SetRequest(HttpRequestMessage request) => actualRequest = request;
+            InitializeMocks(httpResponseMessage, SetRequest);
+
+            HttpClient httpClient = new HttpClient(_handlerMock.Object);
+
+            RegisterService target = new RegisterService(
+                httpClient,
+                _contextAccessor.Object,
+                _accessTokenGenerator.Object,
+                _generalSettings.Object,
+                _platformSettings.Object,
+                new Mock<ILogger<RegisterService>>().Object);
+
+            // Act
+            int actual = await target.PartyLookup("897069650", null);
+
+            // Assert
+            Assert.Equal(expected, actual);
+            _handlerMock.VerifyAll();
+        }
+
+        [Fact]
+        public async Task PartyLookup_ResponseIsNotSuccessful_PlatformExceptioThrown()
+        {
+
+            // Arrange
+
+            HttpResponseMessage httpResponseMessage = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.NotFound,
+                Content = new StringContent(string.Empty)
+            };
+
+            HttpRequestMessage actualRequest = null;
+            void SetRequest(HttpRequestMessage request) => actualRequest = request;
+            InitializeMocks(httpResponseMessage, SetRequest);
+
+            HttpClient httpClient = new HttpClient(_handlerMock.Object);
+
+            RegisterService target = new RegisterService(
+                httpClient,
+                _contextAccessor.Object,
+                _accessTokenGenerator.Object,
+                _generalSettings.Object,
+                _platformSettings.Object,
+                new Mock<ILogger<RegisterService>>().Object);
+
+            // Act & Assert
+            try
+            {
+                int actual = await target.PartyLookup("16069412345", null);
+            }
+            catch (PlatformHttpException e)
+            {
+                Assert.NotNull(e);
+            }
+            catch (Exception)
+            {
+                throw new XunitException("Wrong exception type thrown.");
+            }
```

Comment:
This extra catch is probably unnecessary. If there is an exception different from PlatformException the test will fail as the exception will pass out of the method and be caught by the test runner.